File: Solutions/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public int GetDecimalValue(ListNode head) {
        int res = head.val;
        head = head.next;
        while(head != null){
            res <<= 1;
            res += head.val;
            head = head.next;
        }
        
        return res;
    }
}

//Third fastest
public class Solution {
    public int GetDecimalValue(ListNode head) {
        var num = -1;
        var current = head;
        while(current != null)
        {
            num = num == -1 ? current.val : num * 2 + current.val;
            current = current.next;
        }
        return num;
    }
}

//Second fastest
public class Solution {
    public int GetDecimalValue(ListNode head) {
        int result = 0;
        for(ListNode ptr = head; ptr != null; ptr = ptr.next){
            result = result*2 + ptr.val;
        }
        return result;
    }
}

//Fastest
public class Solution {
    public int GetDecimalValue(ListNode head) {
        var num = -1;
        var current = head;
        while(current != null)
        {
            num = num == -1 ? current.val : num * 2 + current.val;
            current = current.next;
        }
        return num;
    }
}

File: Solutions/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.py
class Solution:
    def getDecimalValue(self, head):
        s = 0
        while head:
            s = 2*s + head.val
            head = head.next
        return s

#Third fastest
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        num=head.val
        while head.next:
            num=num*2+head.next.val
            head=head.next
        return num

#Second fastet
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        ans = ''
        while head:
            ans += str(head.val)
            head = head.next
        return int(ans, 2)

#Fastest
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        number = 0
        length = 0
        copy_head = head
        while head.next != None:
            length += 1
            head = head.next
        check_number = []
        while(length>0): 
            check_number.append([copy_head.val, 2**length])
            number += copy_head.val*(2**length)
            length -= 1
            copy_head = copy_head.next
        number += copy_head.val*(2**length)

        return number